core/lib/src/rocket.rs

```diff
@@ -703,54 +703,49 @@ impl Rocket {
         Ok(self)
     }
 
-    /// Starts the application server and begins listening for and dispatching
-    /// requests to mounted routes and catchers. Unless there is an error, this
-    /// function does not return and blocks until program termination.
-    ///
-    /// # Error
-    ///
-    /// If there is a problem starting the application, a [`LaunchError`] is
-    /// returned. Note that a value of type `LaunchError` panics if dropped
-    /// without first being inspected. See the [`LaunchError`] documentation for
-    /// more information.
+    /// Similar to `launch()`, but using a custom Tokio runtime and returning
+    /// a `Future` that completes along with the server. The runtime has no
+    /// restrictions other than being Tokio-based, and can have other tasks
+    /// running on it.
     ///
     /// # Example
     ///
     /// ```rust
+    /// // This gives us the default behavior. Alternatively, we could use a
+    /// // `tokio::runtime::Builder` to configure with greater detail.
+    /// let runtime = tokio::runtime::Runtime::new();
+    ///
     /// # if false {
-    /// rocket::ignite().launch();
+    /// let fut = rocket::ignite().spawn_on(&runtime).expect("error launching server");
+    /// fut.then(|result| {
```

Comment:
This assert will never be reached (because futures are lazy): something like `runtime.block_on(fut)` would be needed in order to accomplish that. It happens to be the case that servers cannot (yet) terminate successfully either, but we can fix that aspect of this example in #1079.

Reply:
Futures being lazy keeps coming back to bite me. I need to stop relying on my JavaScript background 😁 … It would certainly make sense to tack a `.shutdown` between the two.